pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java

```diff
@@ -283,7 +283,11 @@ private static Object[] canonicalizeRow(Object[] row, DataSchema dataSchema) {
     for (int colId = 0; colId < row.length; colId++) {
       Object value = row[colId];
       if (value != null) {
-        resultRow[colId] = dataSchema.getColumnDataType(colId).convert(value);
+        if (dataSchema.getColumnDataType(colId) == DataSchema.ColumnDataType.OBJECT) {
+          resultRow[colId] = value;
+        } else {
+          resultRow[colId] = dataSchema.getColumnDataType(colId).convert(value);
+        }
```

Comment:
why do we need the additional if check here? i thought dataSchema.convert has already handled OBJECT?

Reply:
good catch, this was from a previous iteration of the PR that was leftover